src/Google.Datastore.V1Beta3/ArrayValuePartial.cs

```diff
@@ -0,0 +1,346 @@
+        /// <summary>
+        /// Converts an array to an <see cref="ArrayValue"/> instance. Each value within <paramref name="values"/> is converted
+        /// to a <see cref="Value"/> using the individual implicit conversion.
+        /// </summary>
+        /// <param name="values">The array to convert.</param>
+        /// <returns>An <see cref="ArrayValue"/> with converted values, or <c>null</c> if <paramref name="values"/> is null.</returns>
+        public static implicit operator ArrayValue(DateTime[] values) => ToArrayValue(values, x => x);
+        /// <summary>
+        /// Converts an array to an <see cref="ArrayValue"/> instance. Each value within <paramref name="values"/> is converted
+        /// to a <see cref="Value"/> using the individual implicit conversion. Any null input values are converted
+        /// to instances of <see cref="Value"/> with a kind of Null.
+        /// </summary>
+        /// <param name="values">The array to convert.</param>
+        /// <returns>An <see cref="ArrayValue"/> with converted values, or <c>null</c> if <paramref name="values"/> is null.</returns>
+        public static implicit operator ArrayValue(DateTime?[] values) => ToArrayValue(values, x => x);
+        /// <summary>
+        /// Converts an array to an <see cref="ArrayValue"/> instance. Each value within <paramref name="values"/> is converted
+        /// to a <see cref="Value"/> using the individual implicit conversion.
+        /// </summary>
+        /// <param name="values">The array to convert.</param>
+        /// <returns>An <see cref="ArrayValue"/> with converted values, or <c>null</c> if <paramref name="values"/> is null.</returns>
+        public static implicit operator ArrayValue(DateTimeOffset[] values) => ToArrayValue(values, x => x);
+        /// <summary>
+        /// Converts an array to an <see cref="ArrayValue"/> instance. Each value within <paramref name="values"/> is converted
+        /// to a <see cref="Value"/> using the individual implicit conversion. Any null input values are converted
+        /// to instances of <see cref="Value"/> with a kind of Null.
+        /// </summary>
+        /// <param name="values">The array to convert.</param>
+        /// <returns>An <see cref="ArrayValue"/> with converted values, or <c>null</c> if <paramref name="values"/> is null.</returns>
+        public static implicit operator ArrayValue(DateTimeOffset?[] values) => ToArrayValue(values, x => x);
+        /// <summary>
+        /// Converts an array to an <see cref="ArrayValue"/> instance. Any null input values are converted
+        /// to instances of <see cref="Value"/> with a kind of Null.
+        /// </summary>
+        /// <param name="values">The array to convert.</param>
+        /// <returns>An <see cref="ArrayValue"/> with converted values, or <c>null</c> if <paramref name="values"/> is null.</returns>
+        public static implicit operator ArrayValue(Value[] values) => ToArrayValue(values, x => x);
```

Comment:
No, it's not valid to have a null reference in a repeated field in proto, ever. The conversion from a `Value` with a kind of Null to a null reference going the other way is a little more debatable, but I've done it for consistency with everything else.